acestep/ui/gradio/events/training/lora_training.py

```
@@ -0,0 +1,304 @@
+        if training_failed:
+            final_msg = f"{failure_message}\nElapsed: {_format_duration(total_time)}"
+            logger.warning(final_msg)
+            log_lines.append(failure_message)
+            yield final_msg, "\n".join(log_lines[-15:]), final_plot, training_state
+            return
+        completion_msg = f"✅ Training completed! Total time: {_format_duration(total_time)}"
+        logger.info(completion_msg)
+        log_lines.append(completion_msg)
+        yield completion_msg, "\n".join(log_lines[-15:]), final_plot, training_state
+
+    except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)